packages/@aws-cdk/aws-rds/lib/instance.ts

```diff
@@ -1032,6 +1037,9 @@ export class DatabaseInstanceFromSnapshot extends DatabaseInstanceSource impleme
         username: credentials.username,
         encryptionKey: credentials.encryptionKey,
         excludeCharacters: credentials.excludeCharacters,
+        // Always replace secret when customization options are changed
+        // (MasterUsername is not specified for an instance created from a snapshot)
+        overrideLogicalId: true,
```

Comment:
Yes because `MasterUsername` is not specified when creating from a snapshot so there's no replacement risk here. It will update the credentials though (both the secret and the database), is this problematic?